### ISSUE TITLE
Forbid local variables that are referenced only once

### ISSUE THREAD

**Issue by dreamflasher** [wemake-services/wemake-python-styleguide#2338]
# Rule request

## Thesis

Instead of:

```
cropped_image = input[a:b, c:d]
rotated_image = rotate(cropped_image)
```

do 

```
rotated_image = rotate(input[a:b, c:d])
```

## Reasoning
My opinion is that the added variable adds unnecessary complexity. But I could also see the argument that if the variable is named well it is part of the documentation/make the code better understandible. What are your thoughts?

**Comment by sobolevn** [comment]
Thanks a lot for the idea! 

But, I belive that this is the opposite of what I recommend. Why?
1. Extra names add extra context, it is better for readability
2. Sometimes you have to split complex expressions into simplier ones and add intermediate variables. This allows us to keep exression complexity at min
3. This might contains lots of corner cases and false positives

I am going to close this one, but there are several things you can do:
1. Propose another ideas, this is a great contribution!
2. Implement this check yourself in a different plugin and you can use together with WPS!

**Comment by dreamflasher** [comment]
Thank you for your feedback. This makes sense.
